### DISCUSSION TITLE
How Can I Use SI Units for Send_Setpoint Commands?

### DISCUSSION THREAD

**Discussion by eli-pratt** [bitcraze/discussions#1503] [Q&A]
Hello!

I am working on implementing an RL-trained controller on the Crazyflie, and I want to be able to use SI units for setpoint commands. I see that an SI unit conversion was implemented into the code here: https://github.com/bitcraze/crazyflie-firmware/issues/537 and that the Lee controller and Brescianini both use the controlmodeForceTorque mode. However, I am unsure how to use this while sending setpoint commands. How can I send setpoint commands that use SI thrust and torque?

**Comment by knmcguire** [comment]
Hi,

So just a clarification that I need to know first, are you looking to implement the controller offboard or onboard of the crazyflie? Because both these approaches have very different strategies that need to be implemented.

**Reply by eli-pratt** [reply]
Hello!

It will be an offboard controller. We are using mujoco menagerie as the simulation environment to train the RL algorithm, which uses torque and thrust as control input.

Thanks!

**Comment by gemenerik** [comment]
If I understand the code and discussions correctly these controllers do not receive thrust and torque setpoints, but instead output them.  `powerDistributionForceTorque` turns resulting thrust and torque into PWM for the motors.
